pxtblocks/fields/field_melodySandbox.ts

```diff
@@ -530,49 +556,167 @@ export class FieldCustomMelody<U extends FieldCustomOptions> extends Blockly.Fie
-        const cellRect = pxsim.svg.child(cellG, "rect", {
-            'cursor': 'pointer',
-            'width': FieldCustomMelody.CELL_WIDTH,
-            'height': FieldCustomMelody.CELL_WIDTH,
-            'stroke': 'white',
-            'data-x': x,
-            'data-y': y,
-            'rx': FieldCustomMelody.CELL_CORNER_RADIUS
-        }) as SVGRectElement;
+    private attachEventHandlersToMatrix() {
+        if ((this.sourceBlock_.workspace as any).isFlyout) return;
 
-        // add appropriate class so the cell has the correct fill color
-        if (this.melody.getValue(x, y)) pxt.BrowserUtils.addClass(cellRect, pxtmelody.getColorClass(x));
-        else pxt.BrowserUtils.addClass(cellRect, "melody-default");
+        this.elt.addEventListener("keydown", this.keyHandler.bind(this));
+        this.elt.addEventListener("focus", (event) => {
+            if (!this.selected) {
+                return;
+            }
+            this.setFocusIndicator(
+                this.cells[this.selected[0]][this.selected[1]]
+            );
+        });
+        this.elt.addEventListener("blur", () => {
+            if (this.cells) {
+                this.setFocusIndicator();
+            }
+        });
 
-        if ((this.sourceBlock_.workspace as any).isFlyout) return;
+        for (let x = 0; x < this.numCol; ++x) {
+            for (let y = 0; y < this.numRow; ++y) {
+                this.attachPointerEventHandlersToCell(x,y,this.cells[x][y]);
+            }
+        }
+    }
 
+    private attachPointerEventHandlersToCell(x: number, y: number, cellRect: SVGElement) {
         pxsim.pointerEvents.down.forEach(evid => cellRect.addEventListener(evid, (ev: MouseEvent) => {
-            this.onNoteSelect(x, y);
+            this.onNoteSelect(y, x); // row,col is y,x ordering
+            this.setFocusIndicator();
             ev.stopPropagation();
             ev.preventDefault();
         }, false));
+    }
 
-        this.cells[x][y] = cellRect;
+    private keyHandler(e: KeyboardEvent) {
+        // First handle high level navigation
+        switch(e.code) {
+            case "Enter":
+            case "Space": {
+                const [x, y] = this.selected;
+                this.onNoteSelect(y, x);
+                this.setFocusIndicator(this.cells[x][y]);
+                this.elt.setAttribute('aria-activedescendant', `${this.sourceBlock_.id}:${x}${y}`);
+                return;
+            }
+            case "Escape": {
+                (this.sourceBlock_.workspace as Blockly.WorkspaceSvg).markFocused();
+                return;
+            }
+        }
+        // If there's nothing currently selected, the first keypress just creates the cursor
+        if (!this.selected) {
+            this.selected = [0,this.getMelodyNote(0) ?? 0];
+        } else {
+            const [x, y] = this.selected;
+            const ctrlCmd = pxt.BrowserUtils.isMac() ? e.metaKey : e.ctrlKey;
+            switch(e.code) {
+                case "KeyW":
```

Comment:
We no longer need to support the old keyboard navigation keys (WASD) so these can be removed.